setup.py

```diff
@@ -170,7 +169,12 @@
 
 
 # pylint: disable=W0122
-exec(compile(open(SALT_VERSION).read(), SALT_VERSION, "exec"))
+if os.path.exists(SALT_VERSION_HARDCODED):
+    with open(SALT_VERSION_HARDCODED, encoding="utf-8") as rfh:
+        SALT_VERSION = rfh.read().strip()
```

Comment:
Should we allow for comments in the _version.txt file ?
Could see someone adding comments, if not us, for example: the current _version.py has the auto-generated comment, which would be good to add, letting it be well-know that it is an auto-generated file.
```
[root@Unknown david]# cat /usr/lib/python3.6/site-packages/salt/_version.py
# This file was auto-generated by salt's setup

from salt.version import SaltStackVersion

__saltstack_version__ = SaltStackVersion(3005, 1, None, 0, '', 0, 0, 'g6226b9cb58')
```

Reply:
> Should we allow for comments in the _version.txt file 

I'd say no. Let's not complicate it again.
It should just contain the version string.

Reply:
Need to document what format that should be somewhere, so someone doesn't write it incorrectly.
Would be good to have test to ensure if it written correctly too, so no one messes up later

Reply:
> Need to document what format that should be somewhere, so someone doesn't write it incorrectly. Would be good to have test to ensure if it written correctly too, so no one messes up later

No one should write it, that's Salt's job. And the version format is Salt's version format. It should already be documented somewhere.

Reply:
A comment even stating that with a heads-up about not mucking with __str__ etc. would be good